warp10/src/main/java/io/warp10/script/mapper/STRICTMAPPER.java

```diff
@@ -79,40 +92,60 @@ public String toString() {
       return sb.toString();
     }
   }
-  
+
   @Override
   public Object apply(WarpScriptStack stack) throws WarpScriptException {
-    Object o = stack.pop(); // maxpoints
-    
+    Object o = stack.pop(); // maxpoints or -maxspan
+
     if (!(o instanceof Number)) {
       throw new WarpScriptException(getName() + " expects a maximum (inclusive) number of values on top of the stack.");
     }
 
 
-    int max = (int) Math.min(Math.max(0L, ((Number) o).longValue()), Integer.MAX_VALUE); // Clamp to positive int
-    
-    o = stack.pop(); // minpoints
-    
+    long max = ((Number) o).longValue();
+
+    o = stack.pop(); // minpoints or -minspan
+
     if (!(o instanceof Number)) {
       throw new WarpScriptException(getName() + " expects a minimum (inclusive) number of values below the top of the stack.");
     }
 
-    int min = (int) Math.min(Math.max(0L, ((Number) o).longValue()), Integer.MAX_VALUE); // Clamp to positive int
+    long min = ((Number) o).longValue();
+
+    boolean isOnCount = true;
+    if (min < 0 || max < 0) { // Either min or max is strictly negative -> timespan definition
```

Comment:
You should support independent min/max specifications, i.e. a least 10 values, but at most 5 minutes